animation_nodes/nodes/color/vertex_color_input.py

```diff
@@ -0,0 +1,84 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode, VectorizedSocket
+from ... data_structures import Color, ColorList, VirtualColorList
+from . c_utils import getVertexColorsFromLoopColors, getPolygonColorsFromLoopColors
+
+colorModeItems = [
+    ("LOOP", "Loop", "Get color of every loop vertex", "NONE", 0),
+    ("VERTEX", "Vertex", "Get color of every vertex", "NONE", 1),
+    ("POLYGON", "Polygon", "Get color of every polgon", "NONE", 2)    
+]
+
+colorLayerIdentifierTypeItems = [
+    ("INDEX", "Index", "Get color layer based on the index", "NONE", 0),
+    ("NAME", "Name", "Get color layer based on the name", "NONE", 1)
+]
+
+class VertexColorInputNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_VertexColorInputNode"
+    bl_label = "Vertex Color Input"
+    errorHandlingType = "EXCEPTION"
+
+    colorMode: EnumProperty(name = "Color Mode", default = "LOOP",
+        items = colorModeItems, update = AnimationNode.refresh)
+
+    colorLayerIdentifierType: EnumProperty(name = "Color Layer Identifier Type", default = "INDEX",
+        items = colorLayerIdentifierTypeItems, update = AnimationNode.refresh)
+
+    useIndexList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+
+        if self.colorLayerIdentifierType == "INDEX":
+            self.newInput("Integer", "Vertex Color Index", "vertexColorIndex")
+        elif self.colorLayerIdentifierType == "NAME":
+            self.newInput("Text", "Name", "colorLayerName")
+
+        self.newOutput("Color List", "Colors", "colors")        
+
+    def draw(self, layout):
+        layout.prop(self, "colorMode", text = "")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "colorLayerIdentifierType", text = "Type")
+
+    def execute(self, object, identifier):
+        if object is None: return
```

Comment:
You should return an empty ColorList in this case.

Reply:
Now, it is returning an empty ColorList.